compiler/rustc_resolve/src/late.rs

```diff
@@ -66,6 +66,15 @@ enum IsRepeatExpr {
     Yes,
 }
 
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+/// Describes whether an `AnonConst` is a type level const arg or
+/// some other form of anon const (i.e. inline consts or enum discriminants)
+enum AnonConstKind {
+    EnumDiscriminant,
+    InlineConst,
+    ConstArg(IsRepeatExpr),
```

Comment:
Should the repeat expr be a separate variant?

Reply:
I think no, repeat expr counts are still const generic arguments except for their special case for back compat reasons (name res doesnt forbid type parameters, but evaluating them must be possible without providing a concrete value of the type parameter). We have a future compat lint currently for using type parameters in repeat expr counts that will hopefully become a hard error and this can just be `ConstArg,` and not have to track if its a repeat expr or not.